File: backend/src/main/java/com/sulaks/TechSpark/service/AuthService.java
package com.sulaks.TechSpark.service;

import com.sulaks.TechSpark.dto.auth.*;


public interface AuthService {
    MessageResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    UserResponse getCurrentUser(String email);
    MessageResponse verifyOtp(VerifyOtpRequest request);
    MessageResponse resendOtp(ResendOtpRequest request);
}

File: frontend/reactapp/src/services/authService.js
import api from "./api";

export const authService = {
  register: async (payload) => (await api.post("/auth/register", payload)).data,
  login: async (payload) => (await api.post("/auth/login", payload)).data,
  getMe: async () => (await api.get("/auth/me")).data,
};
